src/hyp_map.rs

```diff
@@ -109,17 +200,27 @@ impl HypPageTable {
 
 /// A set of global mappings of the hypervisor that can be used to create page tables.
 pub struct HypMap {
-    regions: ArrayVec<HypMapRegion, MAX_HYPMAP_REGIONS>,
+    supervisor_regions: ArrayVec<HypMapFixedRegion, MAX_HYPMAP_SUPERVISOR_REGIONS>,
+    user_regions: ArrayVec<HypMapPopulatedRegion, MAX_HYPMAP_USER_REGIONS>,
 }
 
 impl HypMap {
     /// Create a new hypervisor map from a hardware memory mem map.
-    pub fn new(mem_map: HwMemMap) -> HypMap {
-        let regions = mem_map
+    pub fn new(mem_map: HwMemMap, elf_map: ElfMap) -> HypMap {
```

Comment:
s/elf_map/user_map (or umode_map)?

Reply:
if I have to chose (and not chose `elf_map`, i'll go for `user_map`. :-)

Reply:
Changed to user_map.